pmoves/services/semantic-cache/tokenism.py

```
@@ -0,0 +1,72 @@
+    async def _ensure_connected(self) -> bool:
+        """Lazily connect to NATS. Returns False if unavailable."""
+        if self._nc is not None:
+            return True
+        if not self.settings.tokenism_enabled:
+            return False
+        try:
+            from nats import connect as nats_connect
+
+            self._nc = await nats_connect(self.settings.nats_url)
+            return True
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)